src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -405,62 +445,40 @@ partition_downloader::download_log_with_capped_size(
       .range = {
         .min_offset = model::offset::max(),
         .max_offset = model::offset::min(),
-      }};
-
-    // process from most recent (last one)
-    auto offset_segment_it = offset_map.rbegin();
-    auto offset_end = offset_map.rend();
-    size_t total_size = 0;
-    auto data_found = false;
-
-    // operating on iterators across suspension points is a potentially unsafe
-    // pattern. here the usage is safe since the backing data structure is local
-    // to this coroutine
-    co_await ss::do_until(
-      [&] {
-          return offset_segment_it == offset_end
-                 || // no more segments to process
-                    // or processing this segment would exceed max_size
-                    // and some proper data is ready
-                 (data_found
-                  && (total_size + offset_segment_it->second.size_bytes >= max_size));
       },
-      [&]() -> ss::future<> {
-          auto [_, s] = *offset_segment_it;
-          ++offset_segment_it;
-
-          retry_chain_node fib(&_rtcnode);
-          retry_chain_logger dllog(cst_log, fib);
-          vlog(
-            dllog.debug,
-            "Starting download, base-offset: {}, term: {}, size: {}, fs "
-            "prefix: {}, "
-            "destination: {}",
-            s.base_offset,
-            s.segment_term,
-            s.size_bytes,
-            dlpart.part_prefix,
-            dlpart.dest_prefix);
-          return download_segment_file(s, dlpart).then([&](auto offsets) {
-              if (!offsets || offsets->size_bytes == 0u) {
-                  return;
-              }
-              auto [min_offset, max_offset, size] = *offsets;
-              data_found = true;
-              total_size += size;
-              dloffsets.push_back({
-                .min_offset = min_offset,
-                .max_offset = max_offset,
-              });
-          });
-      });
-    update_downloaded_offsets(std::move(dloffsets), dlpart);
-    if (!data_found) {
-        // The segments didn't have data batches
-        vlog(_ctxlog.debug, "Log segments didn't have data batches");
-        dlpart.range.min_offset = model::offset{0};
-        dlpart.range.max_offset = model::offset{0};
+      .ot_state = ss::make_lw_shared<storage::offset_translator_state>(_ntpc.ntp(), model::prev_offset(start_offset), start_delta())};
```

Comment:
I may be missing something, but I'm surprised we don't have to determine delta_offset_end of the last segment. Otherwise, how does Redpanda know what the next offset should be (especially if we don't download the segments)?